data/src/config/proxy.rs

```diff
@@ -1,37 +1,48 @@
 use serde::Deserialize;
 
 #[derive(Debug, Clone, Deserialize)]
-#[serde(rename_all = "kebab-case")]
-pub enum Kind {
-    Http,
-    Socks5,
-}
-
-#[derive(Debug, Clone, Deserialize)]
-pub struct Proxy {
-    #[serde(rename = "type")]
-    pub kind: Kind,
-    pub host: String,
-    pub port: u16,
-    pub username: Option<String>,
-    pub password: Option<String>,
+pub enum Proxy {
```

Comment:
Let's add `#[serde(rename_all = "snake_case")]` which would be consistent with having snake case keys for all of our config